### PR TITLE
fix: check for layout after nextTick

### PR DESCRIPTION
**Pull request by TheAlexLichter**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/25196

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/plugins/check-if-layout-used.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { nextTick } from "vue"
```
Comment (danielroe) [inline packages/nuxt/src/app/plugins/check-if-layout-used.ts:1]:
```suggestion
import { nextTick } from 'vue'
```

File: packages/nuxt/src/app/plugins/check-if-layout-used.ts [hunk 2]
```diff
@@ -16,7 +17,10 @@ export default defineNuxtPlugin({
       }
     }
     if (import.meta.server) {
-      nuxtApp.hook('app:rendered', checkIfLayoutUsed)
+      nuxtApp.hook("app:rendered", async () => {
+        await nextTick()
+        checkIfLayoutUsed()
+      });
```
Comment (danielroe) [inline packages/nuxt/src/app/plugins/check-if-layout-used.ts:23]: seems like there might be something wrong with our eslint setup 🤔 

```suggestion
      nuxtApp.hook('app:rendered', async () => {
        await nextTick()
        checkIfLayoutUsed()
      })
```

File: packages/nuxt/src/app/plugins/check-if-layout-used.ts [hunk 3]
```diff
@@ -16,7 +17,10 @@ export default defineNuxtPlugin({
       }
     }
     if (import.meta.server) {
-      nuxtApp.hook('app:rendered', checkIfLayoutUsed)
+      nuxtApp.hook('app:rendered', async () => {
+        await nextTick()
+        checkIfLayoutUsed()
+      })
```
Comment (danielroe) [inline packages/nuxt/src/app/plugins/check-if-layout-used.ts:23]: there might be some value in not awaiting the nextTick as that will block `app:rendered`...

```suggestion
      nuxtApp.hook('app:rendered', () => { nextTick(checkIfLayoutUsed) })
```

Comment (manniL) [reply packages/nuxt/src/app/plugins/check-if-layout-used.ts:23]: Makes sense 👍

**Review by danielroe — APPROVED** [review]

**Pull request merged**